pyzabbix/api.py

```diff
@@ -253,9 +260,7 @@ def do_request(self, method, params=None):
         if 'error' in res_json:
             err = res_json['error'].copy()
             err.update({'json': str(request_json)})
-            msg_str = "Error {code}: {message}, {data} while sending {json}"
-            msg = msg_str.format(**err)
-            raise ZabbixAPIException(msg, err['code'])
```

Comment:
So, the old message is not available anymore?

Reply:
Old message just str format of code, message, data and json. You can query them individually now. Like in an example shown.

Actually, we can add logic in exception class: if more than one argument passed, then it means we should return message in old syntax. So message will remain in same format, but if one argument passed and it is type of dict, then we return message in new syntax (just message from result json).

Could be confusing but backward compatible.

@maksim77 what's your thought?

Reply:
I'm pretty ok with removing it. I think it might be better to do a bit breaking change but leave code clean and simple without additional logic. I just don't quite get what do you mean by this:
>That is what I mean. I think this should support both previous and new versions. And you should be able do: